### ISSUE TITLE
Badge content not updating

### ISSUE THREAD

**Issue by Kshitiz-Rimal** [rafalbednarczuk/curved_navigation_bar#94]
It was working fine when using BottomNavigationBar but when I switched to this, the badge content is not updating in realtime.   

`Badge(
                badgeColor: CustomColor.themeSecondary,
                shape: BadgeShape.circle,
                borderRadius: BorderRadius.circular(100),
                child: Icon(Icons.shopping_cart,
                    size: 25,
                    color: widget.selectBottomNavigationIndex == 2
                        ? Colors.white
                        : CustomColor.themeColor
                ),
                badgeContent: Text(cart.toString(),
                    style: TextStyle(color: Colors.white))),`

**Comment by miguel2armas** [comment]
for this problem i solved it using "flutter_bloc",
use the example that is in "flutter_bloc" and you will see that it works to update even when the current menu is selected

**Comment by cybex-dev** [comment]
Can confirm this is an issue when using a `MultiProvider` from the [Provider](https://pub.dev/packages/provider) package

**Problem:** active button does not update with state changes but all other (non active) buttons do update. See #78 too for likely the same issue.

**MCVE:**

```dart
  return MultiProvider(
      providers: [
        StreamProvider<int>(
          create: (context) => Stream<int>.periodic(Duration(seconds: 1), (i) => i),
          initialData: 4,
          updateShouldNotify: (previous, current) {
            print("Count = $current");
            return true;
          },
        ),
      ],
      builder: (context, child) {
        return CurvedNavigationBar(
          height: 60,
          items: [
            Text("${context.watch<int>()}"),
            Text("${context.watch<int>()}"),
            Text("${context.watch<int>()}"), // this one will remain the same, all others will constantly update
            Text("${context.watch<int>()}"),
            Text("${context.watch<int>()}"),
          ],
          onTap: (value) {
            setState(() {
              activeIndex = value;
              widget.onTabChanged(value);
            });
          },
          index: 2,
        );
      },
    );
```

**Comment by cybex-dev** [comment]
See PR #118, hopefully @rafalbednarczuk will merge PR changes soon

**Comment by cybex-dev** [comment]
Best solution for now is to clone/add as submodule to your project, specify the path and make sure the version in `Cache\hosted\pub.dartlang.org` is removed (I found it sometimes doesn't use the updated "local" path)

**Comment by rafalbednarczuk** [comment]
fixed in 1.0.2, please update curved_navigation_bar in pubspec.yaml to 1.0.2
